### PR TITLE
Auto Moderation API

### LINKED ISSUE #36: Implement discords Auto Moderation API
## As a user I want to...
be able to call the [Auto Moderation](https://discord.com/developers/docs/resources/auto-moderation) endpoints

## So that...
I can perform Auto Moderation actions using my discord bot

## Context
Discord provides a rich set of APIs for interacting with channels in a discord server. We should implement all these endpoints. This ticket covers only the Auto Moderation endpoints. https://discord.com/developers/docs/resources/auto-moderation

## Acceptance Criteria
GIVEN a user wants to setup auto moderation rules
THEN the specified auto moderation rule should be applied

### PR DESCRIPTION
**Pull request by nateweisz**
Closes #36 
This PR also fixes and adds the rest of the missing models for auto moderation!

### CODE REVIEW

**Review by surajkumar — APPROVED** [review]

File: src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationActionMetadata.java [hunk 1]
```diff
@@ -5,6 +5,10 @@
...
-        @JsonProperty("channel_id") long channelId,
-        @JsonProperty("duration_seconds") int durationSecond,
-        @JsonProperty("custom_message") String customMessage) {}
+        @JsonProperty("channel_id")
+                long channelId, // only preset when action type is SEND_ALERT_MESSAGE
+        @JsonProperty("duration_seconds")
+                int durationSecond, // only present when action type is TIMEOUT
+        @JsonProperty("custom_message")
+                String customMessage) {} // only present when action type is BLOCK_MESSAGE and is
+                                          // nullable
```
Comment (Taz03) [inline src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationActionMetadata.java:14]: rather just link the discord doc

File: src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationRuleObject.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.javadiscord.jdi.internal.models.guild;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+// TODO: relook at how AutoModeration handler's are made and possibly redo them (alot was missing)
```
Comment (Taz03) [inline src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationRuleObject.java:8]: did u completed this todo or is this for future dev?

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationRuleObject.java (hunk 2 above)
Comment (nateweisz) [reply src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationRuleObject.java:8]: I did do that, just forgot to remove

File: src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationActionMetadata.java (hunk 1 above)
Comment (nateweisz) [reply src/main/java/com/javadiscord/jdi/internal/models/guild/AutoModerationActionMetadata.java]: Done!

**Review by Taz03 — APPROVED** [review]

**Review by surajkumar — APPROVED** [review]

**Pull request merged**